### DISCUSSION TITLE
How to auto dragable?

### DISCUSSION THREAD

**Discussion by nurullah7733** [nerdyman/react-compare-slider#122] [Q&A]
how to auto move handler with out mouse click or hover. its posible?

**Comment by nerdyman** [comment]
Hi @nurullah7733, you can use [`useReactCompareSliderRef`](https://react-compare-slider.vercel.app/?path=/docs/docs-usereactcomparesliderref--docs) to programatically set the position of the slider.

The [transition demo](https://react-compare-slider.vercel.app/?path=/story/demos--transition) uses `useReactCompareSliderRef` to set the position of the slider when it mounts. You can see the code for the demo by clicking on the **Code** tab at the bottom .

![image](https://github.com/nerdyman/react-compare-slider/assets/4908432/5e07a329-e4bf-4d7c-bbac-301fa9a09cba)

You can also set the change the `position` prop on the `<ReactCompareSlider />` component to set change the position on demand but `useReactCompareSliderRef` is more performant.

Please mark this ticket as resolved/answered if this solves the issue for you, thanks!

**Comment by nurullah7733** [comment]
I need to like this website section: -> https://choosygraphics.com/ visite and scroll below and get the Why Choose Us section screenshort here auto infinite showing after and before imgs
![5555](https://github.com/nerdyman/react-compare-slider/assets/59159165/b55ef076-f354-4947-b2da-bb493cec5874)

**Comment by nerdyman** [comment]
@nurullah7733 You should be able to achieve that using `useReactCompareSliderRef`, `transition` and `changePositionOnHover`. 

You'd need to set the `transition` property to something slow like 3 seconds and continually flip the position from `0` to `100`. So when it gets `100` set the position to `0` and when it gets to `0` set the position to `100`. You'll also need to set `transition` to `undefined` and stop the flipping of the position when the user puts their pointer over the slider component and then enable `transition` and the position flipping again when the pointer leaves the slider component.

**Comment by nerdyman** [comment]
I've created an example of how to do this. It autoplays, stops on `pointerdown` events. And restarts on `pointerleave`.

https://react-compare-slider.js.org/?path=/story/recipes--autoplay
